src/ZddMessageBundle.php

```diff
@@ -18,6 +18,7 @@ public function configure(DefinitionConfigurator $definition): void
             ->rootNode()
                 ->children()
                     ->scalarNode('serialized_messages_dir')->defaultNull()->end()
+                    ->scalarNode('messenger_serializer')->defaultNull()->end()
```

Comment:
I don't think we need that value to be explicitly set.

We could simply use the default configured by the user which is `@messenger.default_serializer`.

Reply:
Discussed with @Inkod, introduce a dedicated interface and "prepared" services to add ability to switch between Php / Messenger and also a self implemented serializer is possible.

Reply:
I don't understand why you want to do such a thing.

The interface already exists in the component and the default implementation used already exists thanks to the component config.

Reply:
What problem does it solve? I honnestly just don't get it :sweat_smile:

Reply:
The `@messenger.default_serializer` is an encoder/decoder, it handles `array`.

<img width="1078" alt="Capture d’écran 2023-12-24 à 21 52 26" src="https://github.com/Yousign/zdd-message-bundle/assets/13260307/3831853d-a624-405d-b4fa-70dc1bea6d81">

I like the idea of let the user defines a custom serializer, what I suggest is keep what your did (the interface and the `ZddMessagePhpSerializer` and let the user use service decoration, so we avoid the boilerplate of configuration on our side and let Symfony do the magic